lib/src/neu/verifications/contract-mismatcher.ts

```diff
@@ -328,6 +344,76 @@ export class ContractMismatcher {
     );
   }
 
+  private getQueryParamsArraySerializationStrategy(): { comma: boolean } {
+    const comma =
+      this.contract.config.paramSerializationStrategy.query.array === "comma";
+
+    return { comma };
+  }
+
+  private findMismatchOnRequestQueryParams(
+    endpoint: Endpoint,
+    userInputRequest: UserInputRequest
+  ): Result<Mismatch[], Error> {
+    const queryParamsString = url.parse(userInputRequest.path).query || "";
+    const contractQueryParams = endpoint.request!!.queryParams;
+
+    const queryParams = qs.parse(queryParamsString, {
+      ...this.getQueryParamsArraySerializationStrategy()
+    });
+
+    let result;
+    let mismatches: Mismatch[] = [];
+    for (const {
+      name: queryParamName,
+      optional,
+      type: contractQueryParamType
+    } of contractQueryParams) {
+      const requestQueryParam = queryParams[queryParamName];
+
+      // Query parameter is mandatory and hasn't been provided
+      if (typeof requestQueryParam === "undefined" && !optional) {
+        mismatches.push(
+          new Mismatch(
+            `Query parameter "${queryParamName}" is required but hasn't been provided.`
+          )
+        );
+      }
+
+      if (typeof requestQueryParam === "undefined") {
+        continue;
+      }
+
+      // Validate current request query param against contract
+      result = this.findMismatchOnStringContent(
+        { name: queryParamName, value: requestQueryParam },
+        contractQueryParamType
+      );
+
+      queryParams[queryParamName] = null;
```

Comment:
It is unconventional and not clear why we are mutating the `queryParams` object here. Instead can `checkForNonExistingParams` be refactored to be perform all the logic required using the unmodified `queryParams` and `contractQueryParams`?